contracts/Market.sol

```diff
@@ -356,7 +358,44 @@ contract Market is IMarket, Ownable, ERC721 {
 
 		emit Settled(index, _bets[index].payout, result, recipient);
 
-		_burn(index);
+		_burn(uint256(index));
+	}
+
+	/* 
+	 * @dev Reverse a bet. Return the stake to the bettor and the loan to the vault
+	 * @param index The index of the bet
+	 * @param signature Signature from market owner
+	 */
+	function refund(uint64 index, SignatureLib.Signature calldata signature) external {
+		bytes16 REFUND_CMD = 0x726566756e6400000000000000000000; //Bytes for "refund"
+		bytes32 messageHash = keccak256(
+			abi.encodePacked(REFUND_CMD, address(this), index)
+		);
+
+		require(
+			isValidSignature(messageHash, signature) == true,
```

Comment:
can just do `isValidSignature` no? also i feel like these checks should be in a modifier if they aren't already

Reply:
No because you have to generate the messageHash first.